test/test_operations.py

```diff
@@ -309,6 +309,19 @@ def test_get_xla_tensor(self):
     tx = t.select(1, 12)
     self.assertEqual(tx, sx.data.cpu())
 
+  def test_masked_fill_scalar(self):
+
+    def fn(tensor):
+      # Build a mask from the first line of tensor.
+      # Also, make it have the same rank as the original tensor.
+      mask = tensor[0].ge(0.5).unsqueeze(dim=0)
```

Comment:
I can see the problem (the test would crash before the change) but confused about the PR description, it says the problem happens when we have 2 tensors with the same rank. But here, it seems mask's shape is `torch.Size([1, 2, 2])` and tensor's shape is torch.Size([2, 2]).

Reply:
`tensor` is indeed [2, 2]. However, `tensor[0]` is [2]. Which makes mask `[1, 2]` after the `unsqueeze` call.